core/src/main/java/org/elasticsearch/rest/BaseRestHandler.java

```diff
@@ -43,4 +51,74 @@ protected BaseRestHandler(Settings settings) {
         super(settings);
         this.parseFieldMatcher = new ParseFieldMatcher(settings);
     }
+
+    @Override
+    public final void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+        // prepare the request for execution; has the side effect of touching the request parameters
+        final RestChannelConsumer action = prepareRequest(request, client);
+
+        // validate unconsumed params, but we must exclude params used to format the response
+        final List<String> unconsumedParams = request.unconsumedParams();
+        final Set<String> responseParams = responseParams();
+        final Iterator<String> it = unconsumedParams.iterator();
+
+        // this has to use an iterator lest a ConcurrentModificationException will arise
```

Comment:
not sure I get this comment maybe we shouldn't modify the params at all in that way but create a copy set of the unconsumed ones?